File: 104 Maximum Depth of Binary Tree.c
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */
int maxDepth(struct TreeNode* root) {
    int i;
    int dep = 0;
    if(root == NULL){
        return dep;
    }
    i = DFS(root, dep);
    return i;
}

int DFS(struct TreeNode* root, int dep){
    int tmpR;
    int tmpL;
    if(root == NULL){
        return dep;
    }else{
        dep++;
        tmpL = DFS(root->left, dep);
        tmpR = DFS(root->right, dep);
        return tmpR > tmpL ? tmpR : tmpL;
    }
}

File: 104 Maximum Depth of Binary Tree.java
public class LeetCode104 {
    private class TreeNode {
      int val;
      TreeNode left;
      TreeNode right;
      TreeNode() {}
      TreeNode(int val) { this.val = val; }
      TreeNode(int val, TreeNode left, TreeNode right) {
          this.val = val;
          this.left = left;
          this.right = right;
      }
  }

    public int maxDepth(TreeNode root) {
        int depth =0;
        if(root ==null){
            return depth;
        }
        return dfs(root,depth);

    }

    private int dfs(TreeNode root, int depth){
        if(root==null) return depth;
        depth++;
        return Math.max(dfs(root.left,depth), dfs(root.right,depth));
    }
}
